src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,16 +1,112 @@
+                    break;
+                case "K":
+                    firstHand[i] = "13";
+                    break;
+                case "A":
+                    firstHand[i] = "14";
+                    break;
+            }
+            switch (secondHand[i]){
+                case "J":
+                    secondHand[i] = "11";
+                    break;
+                case "Q":
+                    secondHand[i] = "12";
+                    break;
+                case "K":
+                    secondHand[i] = "13";
+                    break;
+                case "A":
+                    secondHand[i] = "14";
+                    break;
+            }
+        }
+        // Find if first hand has pair and what number
+        firstHasPair = firstHand[0].equals(firstHand[1]);
+        secondHasPair = secondHand[0].equals(secondHand[1]);
+
+        firstPairValue = parseInt(firstHand[0]);
+        secondPairValue = parseInt(secondHand[0]);
+
+        int firstHandMax = Math.max(parseInt(firstHand[0]), parseInt(firstHand[1]));
+        int secondHandMax = Math.max(parseInt(secondHand[0]), parseInt(secondHand[1]));
+        int firstHandMin = Math.min(parseInt(firstHand[0]), parseInt(firstHand[1]));
+        int secondHandMin = Math.min(parseInt(secondHand[0]), parseInt(secondHand[1]));
+
+        if (firstHasPair && secondHasPair){
+            if(firstPairValue >= secondPairValue){
+                return firstHandOriginal;
+            } else {
+                return secondHandOriginal;
+            }
+        } else if(firstHasPair){
+            return firstHandOriginal;
+        } else if(secondHasPair){
+            return secondHandOriginal;
+        }  else {
+
+            return new String[0];
+        }
+        /*
+            if(firstHandMax == secondHandMax){
+                if(firstHandMin >= secondHandMin){
+                    return firstHandOriginal;
+                } else{
+                    return secondHandOriginal;
+                }
+            } else if(firstHandMax > secondHandMax){
+                return firstHandOriginal;
+            } else return secondHandOriginal;
+        }
+*/
+
+
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        //  return new String[]{"Replace me", "with something else"};
     }
 
     // Extension 1
     public String[] winningPairFromArray(String[][] hands) {
         // Implement the winningPairFromArray logic here and return the array containing the winning pair to make the tests pass.
+
+        // Started, did not finish
+        for(int row = 0; row < hands.length; row++){
+            for(int col = 0; col < hands[row].length; col++) {
+                hands[row][col] = "";
+
+            }
+        }
+
         // You can replace the following return value with something appropriate
         return new String[]{"Replace me", "with something else"};
     }
```

Comment:
You've done a good job on the core task of finding the winning pair. Your approach to converting face card values and determining pairs is well-implemented. For full marks, you would need to give ext 1 and 2 a shot.